photon-api/src/main/scala/com/linkedin/photon/ml/evaluation/AreaUnderROCCurveEvaluator.scala

```diff
@@ -49,4 +52,25 @@ protected[ml] class AreaUnderROCCurveEvaluator(
    * @return True if the first score is better than the second
    */
   override def betterThan(score1: Double, score2: Double): Boolean = score1 > score2
+
+  /**
+   * Method used to define equality on multiple class levels while conforming to equality contract. Defines under
+   * what circumstances this class can equal another class.
+   *
+   * @param other Some other object
+   * @return Whether this object can equal the other object
+   */
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[AreaUnderROCCurveEvaluator]
+
+  /**
+   * Compares two [[AreaUnderROCCurveEvaluator]] objects.
+   *
+   * @param other Some other object
+   * @return True if the both models conform to the equality contract and have the same model coefficients, false
+   *         otherwise
+   */
+  override def equals(other: Any): Boolean = other match {
+    case that: AreaUnderROCCurveEvaluator => that.canEqual(this) && super.equals(that)
```

Comment:
This pattern makes sense in the Java world, but IMO it's unidiomatic here in Scala. Matching on the evaluator type already guarantees that we won't call `super.equals` unless the other object is the type we expect. Also, from a performance perspective, this is doing a redundant type check at runtime -- first within the match and then again in `canEqual`.